File: appli-android/app/src/main/java/com/example/sportapp/designsystem/common_components/EmptyListRow.kt
package com.example.sportapp.designsystem.common_components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.sportapp.designsystem.theme.appColors
import com.example.sportapp.designsystem.theme.blueMedium

/**
 * Ligne d'état vide d'une liste : texte italique sur fond [bgRecessed], avec
 * une icône optionnelle. Canonique partagé — remplace les ex-doublons partiels
 * `RoutineTaskEmptyRow` (avec icône) et `SessionEmptyPhaseRow` (sans) (R8).
 * Quand [iconRes] est fourni : icône + texte centrés ; sinon : texte aligné à
 * gauche avec un padding horizontal.
 */
@Composable
fun EmptyListRow(
    text: String,
    modifier: Modifier = Modifier,
    iconRes: Int? = null,
    backgroundColor: Color = appColors.bgRecessed,
    contentColor: Color = blueMedium,
    fontSize: TextUnit = 14.sp,
    verticalPadding: Dp = 4.dp,
) {
    Box(
        modifier = modifier
            .fillMaxWidth()
            .padding(vertical = verticalPadding)
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .height(44.dp)
                .background(backgroundColor, shape = MaterialTheme.shapes.small)
                .then(if (iconRes == null) Modifier.padding(horizontal = 12.dp) else Modifier),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = if (iconRes != null) Arrangement.Center else Arrangement.Start
        ) {
            if (iconRes != null) {
                Icon(
                    painter = painterResource(iconRes),
                    contentDescription = null,
                    tint = contentColor.copy(alpha = 0.5f),
                    modifier = Modifier.size(18.dp),
                )
                Spacer(modifier = Modifier.width(8.dp))
            }
            Text(
                text = text,
                color = contentColor,
                fontSize = fontSize,
                fontWeight = FontWeight.Medium,
                fontStyle = FontStyle.Italic
            )
        }
    }
}

File: appli-web/src/app/designsystem/common_components/empty-list-row.ts
import { ChangeDetectionStrategy, Component, input } from '@angular/core';
import { AppIcon } from '@designsystem/icons/app-icon';

/**
 * Ligne d'état vide — miroir de EmptyListRow.kt : texte italique Medium sur fond bgRecessed,
 * boîte 44px (radius 8). Avec icône (18px, alpha 0.5) → icône + texte centrés ; sans icône →
 * texte aligné à gauche (padding horizontal 12px). contentColor default blueMedium.
 */
@Component({
  selector: 'app-empty-list-row',
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [AppIcon],
  template: `
    <div class="elr" [style.padding-block.px]="verticalPadding()">
      <div class="elr__inner" [class.elr__inner--icon]="!!icon()" [style.background]="backgroundColor()">
        @if (icon()) {
          <app-icon class="elr__icon" [name]="icon()" [size]="18" [color]="contentColor()" />
        }
        <span class="elr__text" [style.color]="contentColor()" [style.font-size.px]="fontSize()">{{ text() }}</span>
      </div>
    </div>
  `,
  styles: [
    `
      .elr {
        width: 100%;
      }
      .elr__inner {
        display: flex;
        align-items: center;
        justify-content: flex-start;
        width: 100%;
        height: 44px;
        border-radius: var(--radius-md);
        padding: 0 var(--space-3);
        box-sizing: border-box;
      }
      .elr__inner--icon {
        justify-content: center;
        padding: 0;
      }
      .elr__icon {
        opacity: 0.5;
        margin-right: var(--space-2);
      }
      .elr__text {
        font-weight: var(--font-weight-medium);
        font-style: italic;
      }
    `,
  ],
})
export class EmptyListRow {
  readonly text = input('');
  /** Ligature Material Symbols ; vide = pas d'icône (texte aligné à gauche). */
  readonly icon = input('');
  readonly backgroundColor = input('var(--app-bg-recessed)');
  readonly contentColor = input('var(--c-blue-medium)');
  readonly fontSize = input(14);
  readonly verticalPadding = input(4);
}
